include/lsst/afw/image/MaskedImage.h

```diff
@@ -79,20 +79,17 @@ class MaskedImage : public lsst::daf::base::Persistable,
                     public lsst::daf::base::Citizen {
 public:
     /// shared pointer to the Image
-    typedef typename Image<ImagePixelT>::Ptr ImagePtr;
+    typedef std::shared_ptr<image::Image<ImagePixelT>> ImagePtr;
     /// shared pointer to the Mask
-    typedef typename Mask<MaskPixelT>::Ptr MaskPtr;
+    typedef std::shared_ptr<image::Mask<MaskPixelT>> MaskPtr;
     /// shared pointer to the variance Image
-    typedef typename Image<VariancePixelT>::Ptr VariancePtr;
-    /// shared pointer to a MaskedImage
-    typedef std::shared_ptr<MaskedImage> Ptr;
-    typedef std::shared_ptr<const MaskedImage> ConstPtr;
+    typedef std::shared_ptr<image::Image<VariancePixelT>> VariancePtr;
```

Comment:
I kept anything not strictly matching `Ptr` or `ConstPtr`, because these more complex typedefs seem useful for expressing class-dependent concepts (e.g., do you really want to remember the exact type for a `MaskedImage`'s variance component?)

Do you think we should eliminate all typedefs for `shared_ptr`, no matter what they're called or what their purpose seems to be?